src/Uno.UI.Composition/Composition/ExpressionAnimationParser/AnimationFunctionCallSyntax.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Numerics;
+
+namespace Microsoft.UI.Composition;
+internal class AnimationFunctionCallSyntax : AnimationExpressionSyntax
+{
+	private AnimationExpressionSyntax _identifierOrMemberAccess;
+	private ImmutableArray<AnimationExpressionSyntax> _arguments;
+
+	public AnimationFunctionCallSyntax(AnimationExpressionSyntax identifierOrMemberAccess, ImmutableArray<AnimationExpressionSyntax> arguments)
+	{
+		_identifierOrMemberAccess = identifierOrMemberAccess;
+		_arguments = arguments;
+	}
+
+	public override object Evaluate(ExpressionAnimation expressionAnimation)
+	{
+		if (_identifierOrMemberAccess is AnimationIdentifierNameSyntax identifier)
+		{
+			return EvaluateFromIdentifier(identifier, expressionAnimation);
+		}
+		else if (_identifierOrMemberAccess is AnimationMemberAccessExpressionSyntax memberAccess)
+		{
+			return EvaluateFromMemberAccess(memberAccess, expressionAnimation);
+		}
+
+		throw new InvalidOperationException($"Unexpected type '{_identifierOrMemberAccess.GetType()}'");
+	}
+
+	private object EvaluateFromIdentifier(AnimationIdentifierNameSyntax identifier, ExpressionAnimation expressionAnimation)
+	{
+		var name = (string)identifier.Identifier.Value;
+		if (name == "Abs")
```

Comment:
@dr1rrb Actually I was thinking of refactoring this whole thing by abstracting all this into an `AnimationFunctionSpecification` which will be abstract class with `ParametersLength` and `MethodName` abstract properties, and an `Evaluate` method, and have a static list of all `AnimationFunctionSpecification` that we loop over.

Let me know your thoughts about such a refactoring. If you think it will be unnecessary complexity, I'll refactor to `switch`

Reply:
I'll have some common code for the `Evaluate` method (i.e, throwing exception if number of arguments is not correct). So instead of duplicating it, it will be in the abstract class.

Reply:
The common code can be done by the caller actually. I'll use an interface.